### PR TITLE
[VS Incentives]: feat: group creation fee

### PR DESCRIPTION
**Pull request by p0mvn**
Closes: #6505

## What is the purpose of the change

Adding fee for group creation and tests

## Testing and Verifying

Updated `TestCreateGauge`

## Documentation and Release Note

  - [ ] Does this pull request introduce a new feature or user-facing behavior changes?
  - [ ] Changelog entry added to `Unreleased` section of `CHANGELOG.md`?

Where is the change documented? 
  - [ ] Specification (`x/{module}/README.md`)
  - [ ] Osmosis documentation site
  - [ ] Code comments?
  - [ ] N/A

### CODE REVIEW

File: proto/osmosis/incentives/params.proto [hunk 1]
```diff
@@ -11,4 +12,10 @@ message Params {
   // (day, week, etc.)
   string distr_epoch_identifier = 1
       [ (gogoproto.moretags) = "yaml:\"distr_epoch_identifier\"" ];
+
+  // group_creation_fee is the fee required to create a new group
+  repeated cosmos.base.v1beta1.Coin group_creation_fee = 6 [
```
Comment (p0mvn) [inline proto/osmosis/incentives/params.proto:17]: Note: `6` is changed to `2` in a future PR

File: x/incentives/keeper/gauge.go [hunk 2]
```diff
@@ -246,8 +246,11 @@ func (k Keeper) CreateGroup(ctx sdk.Context, coins sdk.Coins, numEpochPaidOver u
 		return 0, err
 	}
 
-	// TODO: charge fixed fee. Make sure to update method spec and tests.
-	// https://github.com/osmosis-labs/osmosis/issues/6506
+	// Charge group creation fee.
+	groupCreationFee := k.GetParams(ctx).GroupCreationFee
+	if err := k.bk.SendCoinsFromAccountToModule(ctx, owner, types.ModuleName, groupCreationFee); err != nil {
```
[the remark this answers is not in the document]
Comment (p0mvn) [reply x/incentives/keeper/gauge.go]: Fixed bug but the test is added in https://github.com/osmosis-labs/osmosis/pull/6539

File: proto/osmosis/incentives/params.proto (hunk 1 above)
Comment (p0mvn) [reply proto/osmosis/incentives/params.proto]: Fixed

File: x/incentives/keeper/gauge_test.go [hunk 3]
```diff
@@ -679,16 +679,34 @@ func (s *KeeperTestSuite) TestInitGaugeInfo() {
 	}
 }
 
-// Validates that the Group is created as defined by the CreateGroup spec with the
+// Validates that the Group is created as defined bu the CreateGroup spec with the
 // associated 1:1 group Gauge and the correct gauge records relating to the given pools'
 // internal perpetual gauge IDs.
+//
+// The tes structure is that a general shared state setup is performed once at the top.
```
Comment (czarcas7ic) [inline x/incentives/keeper/gauge_test.go:686]:
```suggestion
// Validates that the Group is created as defined by the CreateGroup spec with the
// associated 1:1 group Gauge and the correct gauge records relating to the given pools'
// internal perpetual gauge IDs.
//
// The test structure is that a general shared state setup is performed once at the top.
```

**Review by czarcas7ic — APPROVED** [review]

File: proto/osmosis/incentives/params.proto [hunk 4]
```diff
@@ -11,4 +12,10 @@ message Params {
   // (day, week, etc.)
   string distr_epoch_identifier = 1
       [ (gogoproto.moretags) = "yaml:\"distr_epoch_identifier\"" ];
+
+  // group_creation_fee is the fee required to create a new group
```
[the remark this answers is not in the document]
Comment (p0mvn) [reply proto/osmosis/incentives/params.proto:16]: Tracking as a task under: https://github.com/osmosis-labs/osmosis/issues/6057

**Pull request merged**